screen is just black

Windows 10 home edition, Alacritty-v0.2.5.exe.  I've also tried with 0.2.4 and get the same results.

I suppose this is something environmental - I mean you'd have noticed if this thing just didn't work on Windows! - but I've no idea what.

When I run Alacritty, I just get a black screen, and a cursor.  Typing gives the impression that I might be getting black text on a black background, the cursor looks as though it's moving the right amount.

Activity on this thread prompted me to have another go with the latest version - 0.3.0 RC 3 - and with that I get this interesting-looking error message:

```
C:\downloads\alacritty>Created log file at "C:\\Users\\Standard\\AppData\\Local\\Temp\\Alacritty-9508.log"
[2019-04-05 19:29] [ERROR] Alacritty encountered an unrecoverable error:

        There was an error initializing the shaders: Failed compiling shader at C:\Users\travis\build\jwilm\alacritty/res/text.v.glsl: Vertex shader failed to compile with the following errors:
ERROR: 0:48: error(#162) Wrong operand types no operation ':' exists that takes a left-hand operand of type 'const float' and a right operand of type 'const int' (or there is no acceptable conversion)
ERROR: 0:49: error(#162) Wrong operand types no operation ':' exists that takes a left-hand operand of type 'const float' and a right operand of type 'const int' (or there is no acceptable conversion)
ERROR: error(#273) 2 compilation errors.  No code generated
```

Does that help?

(I suspect that @smdsbz and I are seeing different issues.)

--- a/src/tty/windows/conpty.rs
+++ b/src/tty/windows/conpty.rs
@@ -15,11 +15,11 @@
 use super::{Pty, HANDLE};
 
 use std::i16;
+use std::io::Error;
 use std::mem;
 use std::os::windows::io::IntoRawHandle;
 use std::ptr;
 use std::sync::Arc;
-use std::io::Error;
 
 use dunce::canonicalize;
 use mio_anonymous_pipes::{EventedAnonRead, EventedAnonWrite};

--- a/src/tty/windows/conpty.rs
+++ b/src/tty/windows/conpty.rs
@@ -218,7 +218,7 @@ pub fn new<'a>(
 
     let mut proc_info: PROCESS_INFORMATION = Default::default();
     unsafe {
-        success = CreateProcessW(
+        let process_found = CreateProcessW(
             ptr::null(),
             cmdline.as_ptr() as LPWSTR,
             ptr::null_mut(),
@@ -231,7 +231,9 @@ pub fn new<'a>(
             &mut proc_info as *mut PROCESS_INFORMATION,
         ) > 0;
 
-        assert!(success);
+        if (!process_found) {
+            panic!("Unable to spawn shell: File can not be found in PATH");
+        }
     }
 
     // Store handle to console

--- a/src/tty/windows/conpty.rs
+++ b/src/tty/windows/conpty.rs
@@ -19,6 +19,7 @@ use std::mem;
 use std::os::windows::io::IntoRawHandle;
 use std::ptr;
 use std::sync::Arc;
+use std::io::Error;
 
 use dunce::canonicalize;
 use mio_anonymous_pipes::{EventedAnonRead, EventedAnonWrite};
@@ -155,11 +156,13 @@ pub fn new<'a>(
 
     // Create the appropriately sized thread attribute list.
     unsafe {
-        success =
+        let failure =
             InitializeProcThreadAttributeList(ptr::null_mut(), 1, 0, &mut size as PSIZE_T) > 0;
 
         // This call was expected to return false.
-        assert!(!success);
+        if (!failure) {
+            panic_shell_spawn();
+        }
     }
 
     let mut attr_list: Box<[BYTE]> = vec![0; size].into_boxed_slice();
@@ -183,7 +186,9 @@ pub fn new<'a>(
             &mut size as PSIZE_T,
         ) > 0;
 
-        assert!(success);
+        if (!success) {
+            panic_shell_spawn();
+        }
     }
 
     // Set thread attribute list's Pseudo Console to the specified ConPTY
@@ -198,7 +203,9 @@ pub fn new<'a>(
             ptr::null_mut(),
         ) > 0;
 
-        assert!(success);
+        if (!success) {
+            panic_shell_spawn();
+        }
     }
 
     // Get process commandline
@@ -218,7 +225,7 @@ pub fn new<'a>(
 
     let mut proc_info: PROCESS_INFORMATION = Default::default();
     unsafe {
-        let process_found = CreateProcessW(
+        success = CreateProcessW(
             ptr::null(),
             cmdline.as_ptr() as LPWSTR,
             ptr::null_mut(),
@@ -231,8 +238,8 @@ pub fn new<'a>(
             &mut proc_info as *mut PROCESS_INFORMATION,
         ) > 0;
 
-        if (!process_found) {
-            panic!("Unable to spawn shell: File can not be found in PATH");
+        if (!success) {
+            panic_shell_spawn();
         }
     }
 
@@ -253,6 +260,11 @@ pub fn new<'a>(
         read_token: 0.into(),
         write_token: 0.into(),
     })
+}
+
+// Panic with the last os error as message
+fn panic_shell_spawn() {
+    panic!("Unable to spawn shell: {}", Error::last_os_error());
 }
 
 impl OnResize for ConptyHandle {

